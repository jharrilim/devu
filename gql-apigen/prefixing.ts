export const prefixer = (source: string, name: string) => {
  const prefixedSource = source
    .replace(/:\s*(\[*)(\w+)/gm, (m, p1, p2) => {
      if (['Int', 'Float', 'String', 'Boolean', 'ID'].includes(p2)) {
        return m;
      }
      return `: ${p1}${name}_${p2}`;
    })
    .replace(/type\s+(\w+)/gm, (m, p1) => {
      return `type ${name}_${p1}`;
    });

  let userRoot = '';

  if (prefixedSource.includes(`type ${name}_Query`)) {
    userRoot += `
type Query {
  """
  This is the root query for [${name}](http://localhost:3000/user/${name}).
  """
  ${name}_query: ${name}_Query
}
`;
  }

  if (prefixedSource.includes(`type ${name}_Mutation`)) {
    userRoot += `
type Mutation {
  """
  This is the root mutation for [${name}](http://localhost:3000/user/${name}).
  """
  ${name}_mutation: ${name}_Mutation
}`;
  }
  const combined = `${prefixedSource}\n${userRoot}`;
  return combined;
};
