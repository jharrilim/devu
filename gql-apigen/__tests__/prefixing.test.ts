import { prefixer } from '../prefixing';

const source = `
type User {
  name: String!
  createdAt: String
  age: Int
  friends: [User]
}

type Query {
  hello: String
  person: User
}

type Mutation {
  createUser(name: String!): User
}
`;

describe(prefixer.name, () => {
  it('prefixes', () => {
    const prefixedSource = prefixer(source, 'test', 1);
    expect(prefixedSource).toMatchSnapshot();
  });
});
