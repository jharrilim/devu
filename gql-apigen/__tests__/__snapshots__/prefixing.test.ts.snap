// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prefixer prefixes 1`] = `
"
type test_User {
  name: String!
  createdAt: String
  age: Int
  friends: [test_User]
}

type test_Query {
  hello: String
  person: test_User
}

type test_Mutation {
  createUser(name: String!): test_User
}


type Query {
  \\"\\"\\"
  This is the root query for [test](http://localhost:3000/user/test).
  \\"\\"\\"
  test_query: test_Query
}

type Mutation {
  \\"\\"\\"
  This is the root mutation for [test](http://localhost:3000/user/test).
  \\"\\"\\"
  test_mutation: test_Mutation
}"
`;
